commons/zenoh-shm/src/cleanup.rs

```diff
@@ -0,0 +1,48 @@
+//
+// Copyright (c) 2024 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+
+use static_init::dynamic;
+
+/// A global cleanup, that is guaranteed to be dropped at normal program exit and that will
+/// execute all registered cleanup routines at this moment
+#[dynamic(lazy, drop)]
+pub(crate) static mut CLEANUP: Cleanup = Cleanup::new();
+
+/// An RAII object that calls all registered routines upon destruction
+pub(crate) struct Cleanup {
+    cleanups: lockfree::queue::Queue<Option<Box<dyn FnOnce() + Send>>>,
```

Comment:
It's not clear to me why the `Queue` needs to contain `Option<Box<...>>` and not simply `Box<...>` if  in the `Drop` implementation you say that `// SAFETY: this is safe as cleanup will never have None elements`.

Reply:
lockfree data structure under the hood uses `core::mem::uninitialized` that states:
`// SAFETY: the caller must guarantee that an uninitialized value is valid for `T`.`
and using `Box<...>` I get runtime panic in `uninitialized` at this line:
`intrinsics::assert_mem_uninitialized_valid::<T>();`

Reply:
Then it's worth adding a comment that also help justifying the `unsafe` in the `Drop`.